ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/SDivOperation.java

```diff
@@ -32,12 +32,12 @@ public SDivOperation(final GasCalculator gasCalculator) {
 
   @Override
   public OperationResult executeFixedCostOperation(final MessageFrame frame, final EVM evm) {
-    final Bytes32 value0 = frame.popStackItem();
-    final Bytes32 value1 = frame.popStackItem();
+    final UInt256 value0 = frame.popStackItem();
+    final UInt256 value1 = frame.popStackItem();
     if (value1.isZero()) {
-      frame.pushStackItem(Bytes32.ZERO);
+      frame.pushStackItem(UInt256.ZERO);
     } else {
-      final BigInteger result = value0.toBigInteger().divide(value1.toBigInteger());
+      final BigInteger result = value0.toSignedBigInteger().divide(value1.toSignedBigInteger());
```

Comment:
just out of curiosity, if value0 and value1 are UInt256, why `toSignedBigInteger` for the division?